src/modules/hip/kernel/tensor_mean.hpp

```diff
@@ -0,0 +1,683 @@
+                           tensorPartialSumArr,
+                           gridDim_x * gridDim_y,
+                           tensorMeanArr,
+                           roiTensorPtrSrc);
+    }
+
+    return RPP_SUCCESS;
+}
+
+// Handle I8 datatype
+RppStatus hip_exec_tensor_mean(Rpp8s *srcPtr,
+                               RpptDescPtr srcDescPtr,
+                               Rpp32f *tensorMeanArr,
+                               RpptROIPtr roiTensorPtrSrc,
+                               RpptRoiType roiType,
+                               rpp::Handle& handle)
+{
+    if (roiType == RpptRoiType::LTRB)
+        hip_exec_roi_converison_ltrb_to_xywh(roiTensorPtrSrc, handle);
+
+    int globalThreads_x = (srcDescPtr->w + 7) >> 3;
+    int globalThreads_y = srcDescPtr->h;
+    int globalThreads_z = handle.GetBatchSize();
+    int gridDim_x = (int) ceil((float)globalThreads_x/LOCAL_THREADS_X);
+    int gridDim_y = (int) ceil((float)globalThreads_y/LOCAL_THREADS_Y);
+    int gridDim_z = (int) ceil((float)globalThreads_z/LOCAL_THREADS_Z);
+
+    if ((srcDescPtr->c == 1) && (srcDescPtr->layout == RpptLayout::NCHW))
+    {
+        Rpp32u tensorPartialSumArrLength = gridDim_x * gridDim_y * gridDim_z;
+        Rpp32s *tensorPartialSumArr;
+        tensorPartialSumArr = reinterpret_cast<Rpp32s*>(handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
+        hipMemsetAsync(tensorPartialSumArr, 0, tensorPartialSumArrLength * sizeof(Rpp32s), handle.GetStream());
+        hipLaunchKernelGGL(tensor_sum_pln1_hip,
+                           dim3(gridDim_x, gridDim_y, gridDim_z),
+                           dim3(LOCAL_THREADS_X, LOCAL_THREADS_Y, LOCAL_THREADS_Z),
+                           0,
+                           handle.GetStream(),
+                           srcPtr,
+                           make_uint2(srcDescPtr->strides.nStride, srcDescPtr->strides.hStride),
+                           tensorPartialSumArr,
+                           roiTensorPtrSrc);
+        hipStreamSynchronize(handle.GetStream());
+        hipLaunchKernelGGL(tensor_mean_grid_result_hip,
+                           dim3(1, 1, gridDim_z),
+                           dim3(1024, 1, 1),
+                           0,
+                           handle.GetStream(),
+                           tensorPartialSumArr,
+                           gridDim_x * gridDim_y,
+                           tensorMeanArr,
+                           roiTensorPtrSrc);
+    }
+    else if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NCHW))
+    {
+        Rpp32u tensorPartialSumArrLength = gridDim_x * gridDim_y * gridDim_z * 3;
+        Rpp32s *tensorPartialSumArr;
+        tensorPartialSumArr = reinterpret_cast<Rpp32s*>(handle.GetInitHandle()->mem.mgpu.scratchBufferHip.floatmem);
+        hipMemsetAsync(tensorPartialSumArr, 0, tensorPartialSumArrLength * sizeof(Rpp32s), handle.GetStream());
+        hipLaunchKernelGGL(tensor_sum_pln3_hip,
```

Comment:
You can attempt to template the kernel too using passing multiple template parameters like this:
```
template<typename T, typename U>
__global__ void kernelName(int number, T *data) {
... ... ...
}

hipLaunchKernelGGL(HIP_KERNEL_NAME(kernelName<Rpp32u, d_uint24>), , , , , );
```

Reply:
This wasn't feasible as I had already templated executors